### ISSUE TITLE
Behavior of bottomSheet used as a navigation for stack navigator

### ISSUE THREAD

**Issue by Thiousi** [StefanoMartella/react-native-simple-bottom-sheet#4]
Hello @StefanoMartella 
In my implementation, I use your awesome bottom sheet as a menu.
My app is organized with a stack navigator, inside the bottom sheet, I list the stack screens in touchableOpacity/Text items and link to the pages with: `onPress={() => this.props.navigation.navigate('Accueil')}`
I'm able to go from page to page but have a strange behavior (and ultimately a breaking bug). I've attached a video demonstrating this.
What I observe:
- the screen is flashing when changing screens in the stack
- going back to a screen previously opened, the bottom sheet remains open
- I can close the bottom sheet, but clicking on the menu button I use to toggle it, it returns `TypeError: null is not an object (evaluating 'global.bottomMenu.togglePanel')`

My Menu Button is calling the bottomSheet like so: `global.bottomMenu.togglePanel()`

My BottomSheet is defining its ref like so: `<BottomSheet  ref={ref=>{ global.bottomMenu=ref }}`

Both the Menu button and BottomSheet are separate components.

Any help would be awesome, I'm that close to considering the menu complete for my app 😄 

https://user-images.githubusercontent.com/4822666/114381803-d7e80a80-9b8b-11eb-96fa-98cdd15bc309.mp4

**Comment by StefanoMartella** [comment]
Hi @Thiousi,
About your observations:

1. Do you experience the blink effect only when you navigate from the bottom sheet? I'm using the component inside my application with react-navigation and I do not experience that behaviour; anyway, I'll test it deeply as soon as I can.
2. You could try to add `global.bottomMenu.togglePanel()` inside the onPress callback before navigating the next screen in order to prevent it from being opened when going back.
3. It looks the reference is lost in some way, can you please paste the code that gives you the error? 

Thank you 😄

**Comment by StefanoMartella** [comment]
For the point 1, I'm asking in order to understand if it is related to the bottom sheet or to `react-navigation`; it looks the blink effect is reported in different issues, for example in the link below:
https://github.com/react-navigation/react-navigation/issues/9256

**Comment by Thiousi** [comment]
Hello,
1. I've tried with a simple navigation outside of the bottom sheet and I get the blinking as well. I'll investigate the link you sent. Not an issue with the bottom sheet indeed 😉 
2. Was an easy fix. Thanks for that :)
3. Definitely losing the reference and I don't understand why.
Here is the code for the hamburger menu icon that I use to open the bottom sheet.
I lose the ref after returning to a page and don't understand how. Thanks again for your help!
**EDIT**: I lose the reference every time I return to a page where the menu has previously been opened: with back, opening the menu and clicking on the page, or with a standard navigate outside of the bottom sheet. This doesn't seem to be linked to your extension but any help in maybe how to properly reference the sheet and toggle it from any page would be awesome!

```javascript
import React from 'react';
import { Ionicons } from '@expo/vector-icons'; 
import { TouchableOpacity, View, Text} from "react-native";

import Styles from "../styles/Styles";

export class MenuButton extends React.Component {

    render () {
        return (
            <TouchableOpacity
            style={[Styles.MenuButton]}
            onPress={() => global.bottomMenu.togglePanel()}>
                <View>
                    <Ionicons title="Menu" name="menu-sharp" style={Styles.MenuButtonIcon}/>
                </View>
            </TouchableOpacity>
        )
    }
};
    
export default MenuButton
```

**Comment by StefanoMartella** [comment]
Where do you mount the bottom sheet? Please paste also that slice of code 😄 

To reference the sheet from any page you could try the approach described here:
https://reactnavigation.org/docs/navigating-without-navigation-prop/

Basically you define an helper like this:

```javascript
// BottomSheetHelper.js

import * as React from 'react';

export const bottomSheetRef = React.createRef();

export function togglePanel() {
  bottomSheetRef.current?.togglePanel();
}

```

then in the component in which you mount the bottom sheet you use the reference in the helper:

```javascript
// BottomMenu.js
// imports

import { bottomSheetRef } from './BottomSheetHelper';

export default function BottomMenu() {
  return (
    <Portal>
        <BottomSheet ref={bottomSheetRef}>
           {/* ... */}
        </BottomSheet>
    </Portal>
  );
}
```

and finally in the menu button component:

```javascript
// imports
import * as BottomSheetHelper from './path/to/BottomSheetHelper.js';

export class MenuButton extends React.Component {

    render () {
        return (
            <TouchableOpacity
                style={[Styles.MenuButton]}
                onPress={() => BottomSheetHelper.togglePanel()}>
                <View>
                    <Ionicons title="Menu" name="menu-sharp" style={Styles.MenuButtonIcon}/>
                </View>
            </TouchableOpacity>
        )
    }
};
    
export default MenuButton
```

I haven't  tested it, but you could try with this approach 😉

**Comment by Thiousi** [comment]
Again, thank you so much for your time in trying to help me here!

I have already tried the exact way you mention, and although I do not lose the global ref, the menu won't open when coming back to the page. Nothing displayed in terminal/logs...

If I understand your question well, here is how all of the screens on which the navigation is done work (simplified of course...):
```javascript
<Host>
  <View>
     <MenuButton navigation={navigation} /> // This is the component that toggles the bottom sheet (see above)
     <View>
     </View>
  </View >
  <BottomMenu navigation={navigation} /> // This is the bottom sheet
</Host>
```

I have done some additional testing, adding a minHeight so I can always see the slider. If I do Home > Settings > Home, Open the slider works, clicking on a link doesn't close the slider anymore, so even within the slider, I "lose" the reference to the slider.

Here is the function that my navigation uses to close the panel and navigate to a page: 

```javascript
goTo = (name) => {
   BottomSheetHelper.togglePanel();
   RootNavigation.navigate(name);
}
```
Here, after coming back to a page I already opened the Bottom Sheet on, the togglePanel() doesn't work but the navigate works.
I'm  lost 😆 

**Edit**: added the BottomMenu component also:
```javascript
export class BottomMenu extends React.Component {

	goTo = (name) => {
		BottomSheetHelper.togglePanel();
		RootNavigation.navigate(name);
	}

	render () {
		return (
			<View style = {{}}>
				<Portal>
					<BottomSheet 
						ref={bottomMenuRef} 
						isOpen={false}
						sliderMinHeight={30}
						sliderMaxHeight={Dimensions.get('window').height * 0.8}
						wrapperStyle= {Styles.BottomSheet}
						lineStyle= {Styles.BottomSheetLine}
						withOverlay>
						Content of menu goes here
					</BottomSheet>
				</Portal>
			</View>
		);
	}
};
    
export default BottomMenu
```

**Comment by StefanoMartella** [comment]
You're welcome! 😄 

Have you tried to remove the question mark (if you have it) after the `current` attribute?

```javascript
export function togglePanel() {
  bottomSheetRef.current?.togglePanel(); // Here
}
```

maybe the reference is still null somehow but you don't get any log because of the question mark

**Comment by Thiousi** [comment]
I did just now. The result is essentially the same: works the first time but not the second one: 
`TypeError: null is not an object (evaluating 'bottomMenuRef.current.togglePanel')`

I'm really not sure what is going on... I've never had a disappearing ref before...

Okay after more investigating... my component is unmounting when returning to a page it was previously opened on. 
I added to my BottomMenu component the following and sure enough I got the log in the console:
```javascript
componentWillUnmount() {
        console.log('unmounted menu')
    }
```

Could it be because the component is referenced on each page?

Okay so this had nothing to do with your plugin and all to do with my bad organization of code :)
I was indeed rendering the same component on each page but instead of referencing them in the App I was referencing multiple time (one for each screen).
I have moved my references to the App and now don't have any issues!

Thanks so much!
